openstates/md/bills.py

```diff
@@ -165,10 +170,132 @@ def parse_bill_votes(self, doc, bill):
                 vote.add_source(href)
                 yield vote
 
+    def parse_bill_votes_new(self, doc, bill):
+        elems = doc.xpath('//a')
+
+        # MD has a habit of listing votes twice
+        seen_votes = set()
+
+        for elem in elems:
+            href = elem.get('href')
+            if (href and "votes" in href and href.endswith('pdf') and
+                    ("Senate" in href or "House" in href) and href not in seen_votes):
+                seen_votes.add(href)
+                vote = self.parse_vote_pdf(href, bill)
+                vote.add_source(href)
+                yield vote
+
+    def parse_vote_pdf(self, vote_url, bill):
+
+        filename, response = self.urlretrieve(vote_url)
+
+        text = convert_pdf(filename, type='text').decode()
+        lines = text.splitlines()
+
+        if 'senate' in vote_url:
+            chamber = 'upper'
+        else:
+            chamber = 'lower'
+
+        date_string = lines[0].split('Date:')[1].strip()
+        date = datetime.datetime.strptime(date_string, "%b %d, %Y %I:%M (%p)")
+
+        pageIndex = -1
+        for index, line in enumerate(lines):
+            if 'Yeas' in line and 'Nays' in line:
+                pageIndex = index
+                break
+
+        yes_count = -1
+        no_count = -1
+        other_count = 0
+
+        if pageIndex > 0:
+
+            counts = re.split(r'\s{2,}', lines[pageIndex].strip())
+
+            for count in counts:
+                number, string = count.split(' ', 1)
+                number = int(number)
+
+                if string == 'Yeas':
+                    yes_count = number
+                elif string == 'Nays':
+                    no_count = number
+                else:
+                    other_count += number
+
+        passed = yes_count > no_count
```

Comment:
This will be erroneous/nonsensical if `pageIndex` was never set, right?

Reply:
Yes, but an error will be thrown already as they will be negative, and if they are not negative its very unlikely that it will be erroneous, for that there has to be "Yeas" and "Nays" in one single line, with non sensical number at perfectly those positions, in any other case it will throw error. either when int() is used or when yes_count or no_count is negative.

Reply:
I think it's cleaner to explicitly raise the error, rather than relying on a data quality check so far down the import pipeline (that, who knows, could be removed at some point).

Reply:
+1